### PR TITLE
Update docker

### LINKED ISSUE #3349: Use Alpine official Docker Library PHP image
There is a [official Docker library](https://hub.docker.com/_/php/) available on Docker. Recommend we use that for the following reasons:

- Alpine images are smaller than the current used image. This would trim the size of the osu-web image smaller than it would be normally.

- Almost all PHP runtimes are in the official image, tagged. This would function as intended to the current use case of `osu-web`.

I'll be experimenting on this before I do a PR to propose my change.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/59711286-use-alpine-official-docker-library-php-image?utm_campaign=plugin&utm_content=tracker%2F21853994&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F21853994&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by nanaya**
Fixes #3349.

- uses alpine base image for php
  - docker's official php image seem to be intended for image builder, not end user as it doesn't use system packaged php but instead built from scratch. Any additional php extensions require compiling from scratch. It's fine for building one-time image but otherwise it'll just waste people's time building stuff
  - php 7.2 is in alpine 3.8 which unfortunately doesn't have docker image yet so it's 7.1 for now
- always run php and node under `$UID` user. Requires it to be set (on some shells? It's set by default in bash) and exported from running shell
- merge composer and php-fpm service
- use node 8 because at least node-sass 4.5 required by laravel-mix 1.0.7 doesn't support node 10
- tested on ubuntu 16.04 and arch
  - not extensively tested but at least starts and can render several pages

closes #3350

---

### CODE REVIEW

**Comment by naoey** [comment]
Nginx could do with a `depends_on: php`? Container exits when trying to run it because `osuweb-php` host isn't up yet:
screen shot 2018-07-04 at 18 17 14

On macOS if that has anything to do with anything

**Comment by nanaya** [comment]
@naoey that doesn't seem right. I don't think hostname definition depends on the service to be up first. I tried the other way (php depends on nginx) and it still works.

**Comment by naoey** [comment]
I think that's because php doesn't have any dependency on the nginx service being up?

**Comment by nanaya** [comment]
I mean I added

```
depends_on:
  - nginx
```

To php service and nginx service still starts just fine even though php hasn't started yet.

**Comment by naoey** [comment]
Doesn't seem like you need the dependency if you use service names as the host name. Basically [this](https://github.com/ppy/osu-web/pull/3457/commits/8acf8f84c67899fa9908fa4769e3353341a7ae0f#diff-cc8fc3d61f66ab0eaab40c3389a748b6R23) change.

**Comment by nanaya** [comment]
Oh whoops I tested wrong thing.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,80 +1,62 @@
...
   nginx:
     image: nginx:latest
-    container_name: osuweb-nginx
+    depends_on:
+      - php
     volumes:
       - ./docker/nginx/nginx-osu-next:/etc/nginx/conf.d/default.conf
       - .:/data/osuweb
     ports:
       - "8080:80"
```
[the remark this answers is not in the document]
Comment (nanaya) [reply docker-compose.yml]: just give me a variable name and it may happen ᕕ( ᐛ )ᕗ

**Pull request merged**